### PR TITLE
🐛 Do not exit on time parse errors

### LINKED ISSUE #2995: `vulnmgmt.advisories` fails to parse some Ubuntu advisories
**Describe the bug**
```
cnspec shell container image ubuntu:focal-20220113
→ loaded configuration from /etc/opt/mondoo/mondoo.yml using source default
→ connected to Ubuntu 20.04.3 LTS
...
cnspec> vulnmgmt.advisories
...
parsing time "" as "2006-01-02T15:04:05Z07:00": cannot parse "" as "2006"
vulnmgmt.advisories: no data available
```

**To Reproduce**

1. Run above shell command
2. Note the missing data

**Expected behavior**
Should show advisories.

**Desktop (please complete the following information):**

- cnspec 9.14.0

**Additional context**
It does work with `registry.access.redhat.com/ubi9/ubi:9.2-755`:

```
cnspec> vulnmgmt.advisories
vulnmgmt.advisories: [
  0: vuln.advisory id="RHSA-2023:6745"
  1: vuln.advisory id="RHSA-2023:5763"
  2: vuln.advisory id="RHSA-2023:6699"
  3: vuln.advisory id="RHSA-2023:6679"
  4: vuln.advisory id="RHSA-2023:7361"
  5: vuln.advisory id="RHSA-2023:6698"
....
```

### PR DESCRIPTION
**Pull request by czunker**
Some CVEs do not have a publish or modified date. Do not exit in these case.

Fixes #2995

### CODE REVIEW

**Comment by czunker** [comment]
This prevents the reported problem:
```
cnquery run container image ubuntu:focal-20220113 -c "vulnmgmt.advisories"                                                                                                                                                                                                                                     
! using builtin provider for os
! using builtin provider for os
→ loaded configuration from /etc/opt/mondoo/mondoo.yml using source default
vulnmgmt.advisories: [
  0: vuln.advisory id="USN-6517-1"
  1: vuln.advisory id="USN-5570-2"
  2: vuln.advisory id="USN-5825-1"
  3: vuln.advisory id="USN-5825-2"
...
```

The cost is missing data inside the CVE as long as the underlying data problem isn't fixed:
```
vulnmgmt.cves: [
  0: {
    id: "CVE-2022-28321"
    state: cannot convert primitive with NO type information
    summary: cannot convert primitive with NO type information
    worstScore: audit.cvss score=9.8
    unscored: cannot convert primitive with NO type information
    published: 366 days 
    modified: 366 days 
  }
```

File: providers/os/resources/vulnmgmt.go [hunk 1]
```diff
@@ -133,13 +138,18 @@ func (v *mqlVulnmgmt) populateData() error {
 
 	mqlVulnCves := make([]interface{}, len(vulnReport.Cves))
 	for i, c := range vulnReport.Cves {
-		parsedPublished, err := time.Parse(time.RFC3339, c.PublishedAt)
+		var parsedPublished time.Time
```
Comment (chris-rock) [inline providers/os/resources/vulnmgmt.go]: I think this should be a time pointer and we pass in MQL the pointer, so that the value is properly null.

Comment (czunker) [reply providers/os/resources/vulnmgmt.go]: Good point. Changed it and now the dates show up as null, when empty.

**Comment by czunker** [comment]
```
vulnmgmt.cves: [
  0: {
    id: "CVE-2022-37434"
    state: cannot convert primitive with NO type information
    summary: cannot convert primitive with NO type information
    worstScore: audit.cvss score=9.8
    unscored: cannot convert primitive with NO type information
    published: null
    modified: null
  }

```

**Review by chris-rock — APPROVED** [review]

**Comment by czunker** [comment]
I've created a follow-up for the missing data: https://github.com/mondoohq/cnquery/issues/3005

**Pull request merged**